components/datetime/src/time_zone.rs

```diff
@@ -598,12 +600,23 @@ impl FormatTimeZone for GenericLocationFormat {
         let Some(essentials) = data_payloads.essentials else {
             return Ok(Err(FormatTimeZoneError::MissingZoneSymbols));
         };
+        let Some(primary_zones) = data_payloads.primary_zones else {
+            return Ok(Err(FormatTimeZoneError::MissingZoneSymbols));
+        };
         let Some(exemplar_cities) = data_payloads.exemplar_cities else {
             return Ok(Err(FormatTimeZoneError::MissingZoneSymbols));
         };
 
-        let Some(location) = exemplar_cities.0.get(&time_zone_id) else {
-            return Ok(Err(FormatTimeZoneError::NameNotFound));
+        let location = if let Some(region) = primary_zones.0.get(&time_zone_id) {
+            exemplar_cities
+                .regions
+                .get(&region.into_tinystr().to_unvalidated())
+                .unwrap_or(region.as_str())
+        } else {
+            let Some(city) = exemplar_cities.cities.get(&time_zone_id) else {
+                return Ok(Err(FormatTimeZoneError::NameNotFound));
+            };
+            city
```

Comment:
Suggestion: the mapping from time zone to city/region is deterministic based on the time zone identity, right? It seems more efficient, without having to change the data layout, to just set the strings in `exemplar_cities` to be regions when they need to be regions at datagen time.

The format `VVV` just prints out the city name, though. To keep that working, maybe add a new data struct for the time zones that have a city name but whose default location name is the region. That can be a back-burner issue because `VVV` is not reachable via semantic skeletons.

Reply:
There are two things here:

1) I do want to use the region display names data struct in the future. If we inline the values, there'll be duplicated data. 
2) The `VVV` thing. We can split the `ExemplarCitiesV1` into a base key for this format, and an extended key for the `VVV` format, but as I've realised in this PR there is *so much boilerplate* required to add a datetime key that I don't want to do it right now.

Reply:
I hear you about the duplicated data, but https://github.com/unicode-org/icu4x/issues/3260 isn't really compatible with that model. (Unless we wanted to change ExemplarCities to use a similar model??)

Base and extended ExemplarCitiesV1 is what I had in mind.

Here's the key thing: most users don't want `VVV` formats. Only `VVVV` is reachable from semantic skeletons and ECMA-402. So, we should optimize for `VVVV` by stripping out the strings that aren't reachable.

This is one of our biggest non-experimental data markers (maybe actually the biggest now), so I think it's important that we are careful with it.

Reply:
I'm happy to do that, but this will break the `VVV` format, so I'd want to remove that then.

Reply:
I'll make a PR removing `VVV`.

Reply:
As far as the duplicated data goes... we're not really regressing with this PR. It seems like something we could fix in 3.0 or sooner if we have a clearer picture at that time of what to do.